File: src/main/java/it/giaquinto/springberry/model/log/LogMessageFactory.java
package it.giaquinto.springberry.model.log;

public abstract class LogMessageFactory {
    public static LogMessage direct(final String msg) {
        return new LogMessage(
                msg,
                LogOut.DIRECT
        );
    }

    public static LogMessage verbose(final String msg) {
        return new LogMessage(
                msg,
                LogOut.VERBOSE
        );
    }

    public static LogMessage warning(final String msg) {
        return new LogMessage(
                msg,
                LogOut.WARNING
        );
    }

    public static LogMessage error(final String msg) {
        return new LogMessage(
                msg,
                LogOut.ERROR
        );
    }
}

File: src/main/java/it/giaquinto/springberry/model/log/LogMessageFactory.kt
package it.giaquinto.springberry.model.log

abstract class LogMessageFactory {
    object Generic {
        fun direct(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.DIRECT
            )
        }

        fun verbose(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.VERBOSE
            )
        }

        fun warning(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.WARNING
            )
        }

        fun error(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.ERROR
            )
        }
    }

    object Pin {
        fun d(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.DIRECT
            )
        }

        fun v(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.VERBOSE
            )
        }

        fun w(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.WARNING
            )
        }

        fun e(msg: String): LogMessage {
            return LogMessage(
                msg,
                LogOut.ERROR
            )
        }
    }
}